### ISSUE TITLE
Pointer dereference error when using connection string

### ISSUE THREAD

**Issue by conor0456** [Azure/azure-event-hubs-go#38]
### Expected Behavior
A connection to be established with Eventhub using environment variables: 
  EVENTHUB_CONNECTION_STRING, 
  EVENTHUB_NAMESPACE
  EVENTHUB_NAME

### Actual Behavior
The program fails when trying to find an aadProvider and never reaches the code to find an sasProvider. Stack trace below:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x121eafa]

goroutine 1 [running]:
eventhub/vendor/github.com/Azure/go-autorest/autorest/adal.retry(0x13022c0, 0xc420088f60, 0xc420124100, 0x0, 0x0, 0x12da6ec)
	/Users/conordarcy/Documents/go/src/eventhub/vendor/github.com/Azure/go-autorest/autorest/adal/token.go:718 +0x18a
eventhub/vendor/github.com/Azure/go-autorest/autorest/adal.(*ServicePrincipalToken).refreshInternal(0xc420122000, 0x1304e40, 0xc420016088, 0x12d3089, 0x1c, 0x0, 0x0)
	/Users/conordarcy/Documents/go/src/eventhub/vendor/github.com/Azure/go-autorest/autorest/adal/token.go:653 +0x27f
eventhub/vendor/github.com/Azure/go-autorest/autorest/adal.(*ServicePrincipalToken).RefreshWithContext(0xc420122000, 0x1304e40, 0xc420016088, 0x0, 0x0)
	/Users/conordarcy/Documents/go/src/eventhub/vendor/github.com/Azure/go-autorest/autorest/adal/token.go:581 +0xa5
eventhub/vendor/github.com/Azure/go-autorest/autorest/adal.(*ServicePrincipalToken).Refresh(0xc420122000, 0x35, 0x12d3089)
	/Users/conordarcy/Documents/go/src/eventhub/vendor/github.com/Azure/go-autorest/autorest/adal/token.go:573 +0x43
eventhub/vendor/github.com/Azure/azure-amqp-common-go/aad.(*TokenProviderConfiguration).NewServicePrincipalToken(0xc42011c000, 0x0, 0x0, 0xc420014140)
	/Users/conordarcy/Documents/go/src/eventhub/vendor/github.com/Azure/azure-amqp-common-go/aad/jwt.go:194 +0x5fd
eventhub/vendor/github.com/Azure/azure-amqp-common-go/aad.NewJWTProvider(0xc420057e80, 0x1, 0x1, 0xc420062040, 0xc420057ec0, 0xc420057e40)
	/Users/conordarcy/Documents/go/src/eventhub/vendor/github.com/Azure/azure-amqp-common-go/aad/jwt.go:137 +0x126
eventhub/vendor/github.com/Azure/azure-event-hubs-go.NewHubWithNamespaceNameAndEnvironment(0xc42001c103, 0x18, 0xc42001414e, 0x4, 0x0, 0x0, 0x0, 0xc420010e00, 0xc420057f18, 0x105b3ad)
	/Users/conordarcy/Documents/go/src/eventhub/vendor/github.com/Azure/azure-event-hubs-go/hub.go:139 +0x6f
eventhub/vendor/github.com/Azure/azure-event-hubs-go.NewHubFromEnvironment(0x0, 0x0, 0x0, 0x12329d7, 0xc420057f78, 0xc42007e058)
	/Users/conordarcy/Documents/go/src/eventhub/vendor/github.com/Azure/azure-event-hubs-go/hub.go:207 +0x156
main.main()
	/Users/conordarcy/Documents/go/src/eventhub/main.go:14 +0x32
exit status 2

### Environment
- OS: MacOS
- Go version: go version go1.10.2 darwin/amd64
- Version of Library: 0.3.0

Variables that are set:
EVENTHUB_NAMESPACE
EVENTHUB_NAME
EVENTHUB_CONNECTION_STRING

**Comment by devigned** [comment]
Hi @conor0456, I've opened a PR upstream for go-autorest which will resolve the issue. For the near term, I'm going to short circuit the env creation process favoring the SAS flow over AAD. This should fix your issue and speed up environmental configuration. You should see the PR tomorrow.

If you'd like to get past this right now, you can build a SAS token provider and pass it into the Hub like so:
```go
provider, err := sas.NewTokenProvider(sas.TokenProviderWithEnvironmentVars())
// handle err
hub, err := NewHub("namespace", "hubName", provider)
```

**Comment by devigned** [comment]
@conor0456 this should be fixed up in version `0.3.1`. Please reopen if it is not resolved.
